### PR TITLE
Allow for initializing a PBXProj via a direct path

### PR DESCRIPTION
**Pull request by maxwellE**
### Short description 📝
Allows for a pbxproj path to be initiated with a direct path, also
moves some related code to the PBXProj class

### Solution 📦
Simply moves around the code that already exists and opens some ACLs to allow callers of the library to use this code

### Implementation 👩‍💻👨‍💻
N/A

### CODE REVIEW

File: Sources/XcodeProj/Objects/Project/PBXProj.swift
Comment (fortmarek) [inline Sources/XcodeProj/Objects/Project/PBXProj.swift:98]: this could possibly be another `init` method with different parameters but then you would have bunch of logic in the initializer, so I suppose we can keep it as-is.

**Review by fortmarek — APPROVED** [review]
I don't see why not make this reusable but would wait for @kwridan review before merging 👍

File: Sources/XcodeProj/Objects/Project/PBXProj.swift
Comment (maxwellE) [reply Sources/XcodeProj/Objects/Project/PBXProj.swift:98]: I am up for either option, whichever you prefer

File: Sources/XcodeProj/Objects/Project/PBXProj.swift
Comment (kwridan) [inline Sources/XcodeProj/Objects/Project/PBXProj.swift:64]: I wonder if this has any performance impact on large projects 🤔

Looking at the code it seems the array `[PBXObject]` is only transiently needed to pass to the public initialiser, but later gets re-assembled as `PBXObjects` _(which internally loops over the objects to process them)_ - this seems like an unnecessary roundtrip 😅

We may be able to resolve this by:

- making this a regular initialiser and directly assign the variables from loaded `pbxproj`
- creating another private initialiser that can be provided with `PBXObjects` directly to avoid needing an extra hop of making the `[PBXObject]`

File: Sources/XcodeProj/Objects/Project/PBXProj.swift
Comment (kwridan) [inline Sources/XcodeProj/Objects/Project/PBXProj.swift:60]: nit: worth naming this as `pbxproj` to avoid confusion - there is another [`PBXProject`](https://github.com/tuist/XcodeProj/blob/main/Sources/XcodeProj/Objects/Project/PBXProject.swift) type in the code base which is slightly different 🤐

**Review by kwridan — COMMENTED** [review]
Thanks for the updates @maxwellE

**Review by kwridan — APPROVED** [review]

**Pull request merged**
